sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -495,8 +506,30 @@ function getAppConfigurationOptions(
       tracingPolicy(),
       syncTokenPolicy(syncTokens),
       ...retryPolicies,
-      ...defaults,      
-    ]
+      ...defaults,
+    ],
+    userAgentHeaderName: getUserAgentHeaderName(internalAppConfigOptions)
   };
 }
 
+/**
+ * @ignore
+ * @internal
+ */
+export function getUserAgentHeaderName(internalAppConfigOptions: InternalAppConfigurationClientOptions) {
```

Comment:
This seems like something that would be useful for all core-http based SDKs. Any reason not to include this as a policy in core-http?

Reply:
Yes, I agree.

I did whitelist the header for logging universally in preparation for us making this more wide-spread. I think we're being cautious since we've only been using the operation header name by default now.

Reply:
Issue created to track the larger change of getting this into core: https://github.com/Azure/azure-sdk-for-js/issues/6530

Reply:
Definitely good to localize the usage of this header for now until we know for sure that we can use it widely.  Thanks for filing that issue!